core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1714,10 +1748,11 @@ public void rename(AlluxioURI srcPath, AlluxioURI dstPath) throws FileAlreadyExi
       FileDoesNotExistException, InvalidPathException, IOException, AccessControlException {
     Metrics.RENAME_PATH_OPS.inc();
     long flushCounter = AsyncJournalWriter.INVALID_FLUSH_COUNTER;
-    // Both src and dst paths should lock WRITE_PARENT, to modify the parent inodes for both paths.
-    try (InodePathPair inodePathPair = mInodeTree
-        .lockInodePathPair(srcPath, InodeTree.LockMode.WRITE_PARENT, dstPath,
-            InodeTree.LockMode.WRITE_PARENT)) {
+    // Both src and dst paths use WRITE locks, despite possibly updating the parent inodes. The
+    // modify operations on the parent inodes are thread safe.
+    try (InodePathPair inodePathPair =
+        mInodeTree.lockInodePathPair(srcPath, InodeTree.LockMode.WRITE, dstPath,
+            InodeTree.LockMode.WRITE)) {
```

Comment:
Why is dst locked with WRITE? Don't we want the parent of dst to be READ locked? (And dst is not expected to already exist)